### PR TITLE
Feature/new docs playground

### PR DESCRIPTION
**Pull request by minevala**
## Description
- Add custom playground component which is more accessible and easier to use for new documentation
- Show Syntaxblock for other code blocks (non-playground blocks)

## Related Issue
- https://helsinkisolutionoffice.atlassian.net/browse/HDS-1021

## Motivation and Context
- HDS is using Docz's playground component to provide a live example of components. This PR uses the same component react-live underneath. But it is more customized for HDS needs, accessibility, more apparent code block, etc.

## How Has This Been Tested?
- Locally on dev's machine

## Screenshots (if appropriate):
Screenshot 2021-11-02 at 13 03 59
Screenshot 2021-11-02 at 13 03 39

👉 [Storybook Demo](https://city-of-helsinki.github.io/hds-demo/docsite-playground/elements/components/buttons/)

### CODE REVIEW

**Review by laurakarhu — APPROVED** [review]
So cool!

**Comment by ronijaakkola** [comment]
To me it seems the the code font size is slightly too small. In designs it seems to match the font size of the help text but the implementation does not use the same code font so it might look a bit different.

The code font in the implementation is 13px size. I would try maybe 16px for the Monospace font. If that is too big, then maybe 14px.

image

**Comment by minevala** [comment]
> To me it seems the the code font size is slightly too small. In designs it seems to match the font size of the help text but the implementation does not use the same code font so it might look a bit different.
> 
> The code font in the implementation is 13px size. I would try maybe 16px for the Monospace font. If that is too big, then maybe 14px.
> 
> image

I need to check this. It is using Courier 16px on my laptop 🤔

File: new-site/src/components/Playground.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+              <span className="playground-block-editor-helper" id={helperTextId}>
+                Press Enter to start editing. Press Esc to stop editing.
+              </span>
+            </div>
+            <div className="playground-block-editor-scrollbox" tabIndex={-1}>
+              <LiveEditor
+                key={resetCount}
+                onChange={onChange}
+                style={{
+                  background: 'none',
```
Comment (VilleMiekkoja) [inline new-site/src/components/Playground.js:123]: Should we put these styles to css file, or is it better to inline them here?

File: new-site/src/components/Playground.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+  };
+
+  useEffect(() => {
+    if (viewPortRef.current) {
+      const textArea = getTextArea(viewPortRef.current);
+      textArea.setAttribute('aria-describedby', helperTextId);
+      textArea.setAttribute('tabIndex', '-1');
+      textArea.addEventListener('blur', onTextAreaBlur);
+    }
+  }, [getTextArea, helperTextId, onTextAreaBlur]);
```
Comment (VilleMiekkoja) [inline new-site/src/components/Playground.js:98]: I guess technically the `onTextAreaBlur` can never change in the current setting, as it's wrapped inside a `useCallback` hook that has an empty list as a dependency? But React probably still requires it to be set as a dependency for the useEffect.

File: new-site/src/components/Playground.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                }}
+                textareaId={textAreaId}
+                theme={theme}
+              />
+            </div>
+          </div>
+        </div>
+      </div>
+      <div className="playground-block-buttons">
+        <Button ref={copyButtonRef} variant="secondary" size="small" tabIndex={0} onClick={copy}>
```
Comment (VilleMiekkoja) [inline new-site/src/components/Playground.js:139]: Do we need the `tabIndex={0}` here? Buttons should be focusable by default.

File: new-site/src/components/Playground.scss [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+@import '~hds-design-tokens/lib/all.scss';
+
+.playground-block {
+  --focus-border-width: 2px;
+  margin: var(--spacing-l) 0;
+}
+
+.playground-block-tabs {
+  --tablist-border-color: transparent;
```
Comment (VilleMiekkoja) [inline new-site/src/components/Playground.scss:11]: I guess this overrides the `--tablist-border-color` variable defined in Tabs.module.scss? Does it depend on the css order for which one of them takes precedence over another?

**Comment by VilleMiekkoja** [comment]
Works and looks good! 👍  Added couple minor comments.

File: new-site/src/components/Playground.js (hunk 1 above)
Comment (minevala) [reply new-site/src/components/Playground.js:123]: The LiveEditor component seems to take only style property. It passes those styles to underlying textarea and pre components. I guess it is possible to give the className to parent LiveProvider component and then override the child element styles in css file 🤔

File: new-site/src/components/Playground.js (hunk 2 above)
Comment (minevala) [reply new-site/src/components/Playground.js:98]: - [ ] simplify onTextAreaBlur

File: new-site/src/components/Playground.js (hunk 3 above)
Comment (minevala) [reply new-site/src/components/Playground.js:139]: - [ ] remove unnecessary tabIndexes

File: new-site/src/components/Playground.scss (hunk 4 above)
Comment (minevala) [reply new-site/src/components/Playground.scss:11]: I believe it overrides the root value when the property is given in the element scope 🤔

File: new-site/src/components/Playground.js (hunk 1 above)
Comment (minevala) [reply new-site/src/components/Playground.js:123]: Hmm, it might actually take to className as well. I'll try to use that one

Comment (minevala) [reply new-site/src/components/Playground.js:123]: - [ ] replace live-editor inline styles with classname

**Review by ronijaakkola — APPROVED** [review]
This is beautiful.

**Review by VilleMiekkoja — APPROVED** [review]
Good job with this one! 🥇

**Pull request merged**